http/src/lib.rs

```diff
@@ -306,10 +309,10 @@ fn create_request_processing_future<CC: hyper::client::Connect>(
             })
             .map(|response_chunk| response_chunk.to_vec())
             .then(move |response_result| {
-                response_tx.send(response_result).map_err(|_| {
+                if let Err(_) = response_tx.send(response_result) {
                     warn!("Unable to send response back to caller");
-                    ()
-                })
+                }
+                Ok(())
```

Comment:
The problem was that if a user of this library dropped the `RpcRequest` after it had been submitted to the event loop, but before the result was available. Then the channel for the response result (`response_rx`) would be dropped and this send operation would fail. We should just warn if that happens, not crash the entire event loop, which serve multiple clients and an error in just one of them should not affect the others.

So instead of making the result of `response_tx.send` be returned from this `.then` closure, I just warn if there is an error and then **always** return an `Ok(())`.